pkg/archive/wrap.go

```diff
@@ -22,7 +23,7 @@ import (
 //
 // FIXME: stream content instead of buffering
 // FIXME: specify permissions and other archive metadata
-func Generate(input ...string) (Archive, error) {
+func Generate(input ...string) (io.ReadCloser, error) {
```

Comment:
Since this is returning an `ioutil.NopCloser`, would it be better to return an `io.Reader` and have the caller wrap it if necessary?